src/example6.cpp

```diff
@@ -0,0 +1,61 @@
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <dr/mhp.hpp>
+#include <fmt/core.h>
+
+namespace mhp = dr::mhp;
+
+using T = uint16_t;
+using MDA = dr::mhp::distributed_mdarray<T, 2>;
+
+int main() {
+#ifdef SYCL_LANGUAGE_VERSION
+  mhp::init(sycl::default_selector_v);
+#else
+  mhp::init();
+#endif
+  std::size_t arr_size = 7;
+  const std::size_t pattern_size = 2;
+  const std::size_t radius = pattern_size - 1;
+  std::array slice_starts{radius - 1, radius - 1};
+  std::array slice_ends{arr_size - radius, arr_size - radius};
+
+  auto dist = dr::mhp::distribution().halo(radius);
+  MDA a({arr_size, arr_size}, dist);
+  MDA occurrences_coords({arr_size, arr_size});
+
+  mhp::iota(a, 1);
+  mhp::transform(a, a.begin(), [](auto &&v) { return v % 2; });
+  mhp::fill(occurrences_coords, 0);
+
+  auto a_submdspan =
+      dr::mhp::views::submdspan(a.view(), slice_starts, slice_ends);
+  auto occurrences = dr::mhp::views::submdspan(occurrences_coords.view(),
```

Comment:
Are the two structures - occurrences_coords and occurrences - really necessary? They are only used to present the results.

Reply:
One structure is a distributed_mdarray which is populated with values and the second one is a submdspan which is a view over the array and used by a stencil_for_each. I haven't found out a different way to achieve the same result but I am open for any suggestion

Reply:
'a' is an original array populated with values to search.
'occurrences_coords' is an array filled with 0's. 1 will indicate that the pattern was found at this position (precisely - it is the upper left corner of the found pattern). So you will never fill the very right column and the very bottom row - because the upper left corner of a 2x2 area cannot be there.
Why can't the 'occurrences_coords' array be smaller than 'a' as is the 'occurences' view? The view would not be needed in this case

Reply:
I don't want it to be smaller because:
1) stencil_for_each takes multiple views as arguments and I couldn't find any other way around it (even with a for_each) because this code is executing on the kernel so it is not that simple (from my perspective at least)
2) I want to print it exactly the same size as an 'a' array to make it easier to see how it works comparing with the original array

Reply:
1. _occurrences_coords_ need not to be smaller in fact, still two structures _occurrences_coords_ and _occurrences_ is overkill.
2. Pattern search within lambda uses fixed size of the pattern (2x2). You can reduce number of const parameters - pattern_size must be 2, radius must 1 (if only you do not put a loop into _mdspan_pattern_op_). Please reduce number of vars/consts - personally I'd remove _radius_ and leave _pattern_size_

Reply:
Done, thanks